Core.ApplicationServices/GDPR/Write/DataProcessingRegistrationWriteService.cs

```diff
@@ -330,6 +333,23 @@ private Maybe<OperationError> UpdateBasisForTransfer(DataProcessingRegistration
                 .MatchFailure();
         }
 
+        private Result<DataProcessingRegistration, OperationError> UpdateMainContract(DataProcessingRegistration dpr, Guid? contractId)
+        {
+            if (!contractId.HasValue)
+            {
+                dpr.ResetMainContract();
```

Comment:
you must use the application service in this write service. Look at the other implementations for reference.

Right now, you are bypassing auth rules